gcp-function-http/src/main/java/io/micronaut/gcp/function/http/GoogleFunctionHttpResponse.java

```diff
@@ -118,11 +117,6 @@ public Optional<B> getBody() {
 
     @Override
     public <T> MutableHttpResponse<T> body(@Nullable T body) {
-        if (body instanceof CharSequence) {
-            if (!getContentType().isPresent()) {
-                contentType(MediaType.TEXT_PLAIN_TYPE);
-            }
-        }
```

Comment:
@sdelamo I had to make this change in AWS as well to make the TCK pass...  I took the rule that "tck is king", but it is a breaking change in behaviour (see CookiesSpec, et al)

Reply:
Yes, a controller returning a String should return JSON type 

https://github.com/micronaut-projects/micronaut-core/pull/9314/files

This is `CookieSpec`

```
    @Get("/cookies")
    io.micronaut.http.HttpResponse<String> cookies(@CookieValue String myCookie) {
        return io.micronaut.http.HttpResponse.ok(myCookie)
                    .cookie(Cookie.of("foo", "bar").httpOnly(true).domain("https://foo.com"));
    }
```

I think it is consistent with the TCK behaviour that it should return JSON.